lib/callableFunction.js

```diff
@@ -0,0 +1,81 @@
+'use strict';
+
+var _ = require('lodash');
+var request = require('request');
+
+var _isDatabaseFunc = function(eventTrigger) {
+  return /firebase.database/.test(eventTrigger.eventType);
+};
+
+var _substituteParams = function(resource, params) {
+  var wildcards = resource.match(new RegExp('{[^/{}]*}', 'g'));
```

Comment:
I'd take a look at using `String#replace()` with a function ([see docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace#Specifying_a_function_as_a_parameter)) here -- I think the code would end up being cleaner.